### DISCUSSION TITLE
Using ControlNet Depth to generate pngs with transparency?

### DISCUSSION THREAD

**Discussion by Susanoo1337** [Mikubill/sd-webui-controlnet#323] [Ideas]
Not sure how feasible it would be, but could a function be implemented that would use the depth map(or any other controlnet feature) to extract a foreground object and save a png without putting it back into a full image?

**Comment by cmeka** [comment]
Yes it's possible but sounds like it's outside the scope of this extension and even stable diffusion in general. Would have to be its own extension or gradio tool.

**Comment by brunogcar** [comment]
i would recommend https://github.com/thygate/stable-diffusion-webui-depthmap-script 
if i understood you correctly you want to for example remove a character (foreground) from the scene (background)
it can invert remove background, Invert DepthMap (black=near, white=far), 
you can select different models:  midas and leres, etc
but if you want to do like photoshop, just remove and it would be replaced,you would have to inpaint what was removed

EDIT: i just read the title, transparency AFAIK isnt possible, currently there is no layer for that

**Comment by kft334** [comment]
Not sure if this is what you mean... This is easy to implement and works really well with hed/canny. Not so much with normals though.

Edit: My project is here in case you'd like to try it. It needs A1111 and sd-webui-controlnet installed to use. And a lot of the info on the itch.io page is outdated. The source code is included in the download: https://kfs334.itch.io/prompt-crafter-organizer

![Afro](https://user-images.githubusercontent.com/93904307/221378170-ab577cdd-caa1-4c7f-9ac1-83fd04c9d1b3.png)
![Asset](https://user-images.githubusercontent.com/93904307/221378204-6780f767-e8e5-46c8-b3b9-7a1f679a5c77.png)
![Tractor](https://user-images.githubusercontent.com/93904307/221378212-ed6035a9-77bf-4e61-9b01-bdb32889ec90.png)
![Skateboard](https://user-images.githubusercontent.com/93904307/221378298-a28438f4-633a-44a2-abc3-8a538ca8c2d8.png)
![combined](https://user-images.githubusercontent.com/93904307/221378302-ac8299a2-042b-4c01-bb31-8e952b036fa6.png)

Edit: Works with blender too now. Only takes one click in blender to convert a scene to a ControlNet object with transparency.

![Screenshot 2023-02-26 141919](https://user-images.githubusercontent.com/93904307/221432050-74755dfc-0a9a-4055-b576-4e057b0ef8ad.png)

**Comment by HolidayUser1** [comment]
I've been playing with settings and I am unable to graft in the fifth finger I need for the image. The image is clipart so I don't know how much that effects things. It always comes out 4 fingers...or a mangled mess.
